### ISSUE TITLE
compilation failure

### ISSUE THREAD

**Issue by simevo** [doxygen/doxygen#8304]
```
...
[ 70%] Building CXX object src/CMakeFiles/doxymain.dir/docgroup.cpp.o
cd /tmp/doxygen/build/src && /usr/bin/c++ -DYY_BUF_SIZE=262144 -DYY_READ_BUF_SIZE=262144 -I/tmp/doxygen/qtools -I/tmp/doxygen/libmd5 -I/tmp/doxygen/liblodepng -I/tmp/doxygen/libmscgen -I/tmp/doxygen/libversion -I/tmp/doxygen/vhdlparser -I/tmp/doxygen/src -I/tmp/doxygen/build/generated_src -I/usr/lib/llvm-11/include -Wdate-time -D_FORTIFY_SOURCE=2 -g -O2 -fdebug-prefix-map=/tmp/doxygen=. -fstack-protector-strong -Wformat -Werror=format-security -fvisibility=hidden -fvisibility-inlines-hidden   -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -Wall -Wextra -Wshadow -Wnon-virtual-dtor -Wcast-align -Wunused -Woverloaded-virtual -Wpedantic -Wdouble-promotion -Wformat=2 -Wno-unused-parameter -Wno-sign-conversion -Wno-format-nonliteral -Wno-implicit-fallthrough -std=gnu++14 -o CMakeFiles/doxymain.dir/docgroup.cpp.o -c /tmp/doxygen/src/docgroup.cpp
/tmp/doxygen/src/clangparser.cpp: In member function ‘void ClangTUParser::writeLineNumber(CodeOutputInterface&, const FileDef*, uint)’:
/tmp/doxygen/src/clangparser.cpp:495:22: error: invalid conversion from ‘const Definition*’ to ‘Definition*’ [-fpermissive]
  495 |   Definition *d = fd ? fd->getSourceDefinition(line) : 0;
      |                   ~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                      |
      |                      const Definition*
/tmp/doxygen/src/clangparser.cpp:499:40: error: invalid conversion from ‘const MemberDef*’ to ‘MemberDef*’ [-fpermissive]
  499 |     MemberDef *md = fd->getSourceMember(line);
      |                     ~~~~~~~~~~~~~~~~~~~^~~~~~
      |                                        |
      |                                        const MemberDef*
```

C++ version:
```
/usr/bin/c++ -v
Using built-in specs.
COLLECT_GCC=/usr/bin/c++
COLLECT_LTO_WRAPPER=/usr/lib/gcc/x86_64-linux-gnu/10/lto-wrapper
OFFLOAD_TARGET_NAMES=nvptx-none:amdgcn-amdhsa:hsa
OFFLOAD_TARGET_DEFAULT=1
Target: x86_64-linux-gnu
Configured with: ../src/configure -v --with-pkgversion='Debian 10.2.1-3' --with-bugurl=file:///usr/share/doc/gcc-10/README.Bugs --enable-languages=c,ada,c++,go,brig,d,fortran,objc,obj-c++,m2 --prefix=/usr --with-gcc-major-version-only --program-suffix=-10 --program-prefix=x86_64-linux-gnu- --enable-shared --enable-linker-build-id --libexecdir=/usr/lib --without-included-gettext --enable-threads=posix --libdir=/usr/lib --enable-nls --enable-bootstrap --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --with-default-libstdcxx-abi=new --enable-gnu-unique-object --disable-vtable-verify --enable-plugin --enable-default-pie --with-system-zlib --enable-libphobos-checking=release --with-target-system-zlib=auto --enable-objc-gc=auto --enable-multiarch --disable-werror --with-arch-32=i686 --with-abi=m64 --with-multilib-list=m32,m64,mx32 --enable-multilib --with-tune=generic --enable-offload-targets=nvptx-none=/build/gcc-10-0py4jQ/gcc-10-10.2.1/debian/tmp-nvptx/usr,amdgcn-amdhsa=/build/gcc-10-0py4jQ/gcc-10-10.2.1/debian/tmp-gcn/usr,hsa --without-cuda-driver --enable-checking=release --build=x86_64-linux-gnu --host=x86_64-linux-gnu --target=x86_64-linux-gnu --with-build-config=bootstrap-lto-lean --enable-link-mutex
Thread model: posix
Supported LTO compression algorithms: zlib zstd
gcc version 10.2.1 20201224 (Debian 10.2.1-3)
```

**Comment by simevo** [comment]
also:
```
/home/paolog/Sviluppo/debian/doxygen/doxygen/src/clangparser.cpp:508:27: error: invalid conversion from ‘const MemberDef*’ to ‘MemberDef*’ [-fpermissive]
  508 |       p->currentMemberDef=md;
      |                           ^~
      |                           |
      |                           const MemberDef*
```

**Comment by simevo** [comment]
here is a patch: [aaa.zip](https://github.com/doxygen/doxygen/files/5765408/aaa.zip)

not sure why this code is not covered by your compilation process. probably some config difference...

**Comment by doxygen** [comment]
@simevo building with use_libclang enabled is not yet part of our build matrix. Will be enabled soon to catch such issues.

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.1.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
